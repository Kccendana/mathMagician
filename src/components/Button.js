import PropTypes from 'prop-types';

function Button(props) {
  const { label, className, handleButtonClick } = props;
  return (
    <button type="button" className={className} onClick={handleButtonClick}>
      {label}
    </button>
  );
}

Button.propTypes = {
  label: PropTypes.string.isRequired,
  className: PropTypes.string,
  handleButtonClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  className: '',
};

export default Button;
