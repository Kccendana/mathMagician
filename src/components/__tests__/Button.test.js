import React from "react";
import { render } from "@testing-library/react";
import Button from "../Button";

describe('Button component', () => {
    it('should match the snapshot', () => {
        const handleButtonClick = jest.fn();
        const { container } = render(<Button className="button" type="button" label="Click Me" handleButtonClick={handleButtonClick} />);
        expect(container).toMatchSnapshot();
    });

});