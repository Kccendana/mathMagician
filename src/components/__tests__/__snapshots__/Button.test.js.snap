// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match the snapshot 1`] = `
<div>
  <button
    class="button"
    type="button"
  >
    Click Me
  </button>
</div>
`;
